src/LaunchPadPayHook.sol

```diff
@@ -0,0 +1,122 @@
+//   • An Ownable toggle (setFundingTurnedOff) for the fundingTurnedOff flag.
+//   • The beforePayRecordedWith function manipulates the weight to change number
+//     of tokens received per ETH based on the funding status.
+contract LaunchPadPayHook is IJBRulesetDataHook, Ownable {
+
+    uint256 public immutable minFundingRequired;
+    uint256 public immutable fundingGoal;
+    uint256 public immutable deadline;
+
+    // fundingTurnedOff can be toggled by the owner.
+    bool public fundingTurnedOff;
+
+    IJBTerminalStore public jbTerminalStore;
+
+    constructor(
+        uint256 _minFundingRequired,
+        uint256 _fundingGoal,
+        uint256 _deadline,
+        address _jbTerminalStoreAddress,
+        address owner
+    ) Ownable(owner) {
+        minFundingRequired = _minFundingRequired;
+        fundingGoal = _fundingGoal;
+        deadline = _deadline;
+        jbTerminalStore = IJBTerminalStore(_jbTerminalStoreAddress);
+    }
+
+    function setFundingTurnedOff(bool _fundingTurnedOff) external onlyOwner {
+        fundingTurnedOff = _fundingTurnedOff;
+    }
+
+    function _totalFunding(address terminal, uint256 projectId) internal view returns (uint256) {
+        return jbTerminalStore.balanceOf(terminal, projectId, JBConstants.NATIVE_TOKEN);
+    }
+
+    function beforePayRecordedWith(JBBeforePayRecordedContext calldata context) external view override returns (uint256 weight, JBPayHookSpecification[] memory hookSpecifications) {
+        if (fundingTurnedOff) {
+            revert("Project funding has been turned off.");
+        }
+
+        // Get current funding and the incoming payment amount.
+        uint256 currentFunding = _totalFunding(context.terminal, context.projectId);
+        require(context.amount.token == JBConstants.NATIVE_TOKEN);
+        uint256 paymentAmount = context.amount.value;
+
+        // Define our rates:
+        // Rate values include the 1e18 multiplier, and are doubled (due to 50% project tokens split as noted).
+        uint256 rateTier1 = 4_000_000_000_000_000_000_000; // 2,000 tokens per ETH for funding below minFundingRequired
+        uint256 rateTier2 = 2_000_000_000_000_000_000_000; // 1,000 tokens per ETH for funding between minFundingRequired and fundingGoal
+        uint256 rateTier3 = 1_000_000_000_000_000_000_000; // 500 tokens per ETH for funding above fundingGoal
+
+        weight = 0;
+        uint256 remainingPayment = paymentAmount;
+        uint256 tempFunding = currentFunding;
+
+        // In the case where a payment will bring a project over minFundingRequired or fundingGoal
+        // we need to calculate the weighted average of the rates.
+
+        // Tier 1: Payments that bring total funding up to minFundingRequired.
+        if (tempFunding < minFundingRequired) {
+            if (block.timestamp >= deadline) {
+                revert("Project funding deadline has passed and minimum funding requirement has not been met.");
```

Comment:
I guess here is where we'll want to trigger the "redemptions" cycle

Reply:
correct!